src/meshes/p4est_mesh.jl

```diff
@@ -497,45 +497,319 @@ function p4est_connectivity_from_hohqmesh_abaqus(meshfile, initial_refinement_le
+                                              elements_begin_idx,
+                                              sets_begin_idx)
+    meshfile_p4est_rdy = replace(meshfile_pre_proc,
+                                 "_preproc.inp" => "_p4est_ready.inp")
+    order = 1 # Assume linear elements by default
+
+    # Some useful function-wide variables
+    current_element_type = ""
+    new_line = ""
+
+    open(meshfile_pre_proc, "r") do infile
+        open(meshfile_p4est_rdy, "w") do outfile
+            for (line_index, line) in enumerate(eachline(infile))
+                # Copy header and node data
+                if line_index < elements_begin_idx || line_index >= sets_begin_idx
+                    println(outfile, line)
+                else # Element data
+                    # Check if a new element type/element set is defined
+                    if startswith(line, "*ELEMENT")
+                        # Retrieve element type
+                        current_element_type = match(r"\*ELEMENT, type=([^,]+)", line).captures[1]
+
+                        # Check for linear elements - then we just copy the line
+                        if occursin(linear_quads, current_element_type) ||
+                           occursin(linear_hexes, current_element_type)
+                            println(outfile, line)
+                        else # Quadratic element - replace with linear
+                            order = 2
+                            if occursin(quadratic_quads, current_element_type)
+                                linear_quad_type = "CPS4"
+                                new_line = replace(line,
+                                                   current_element_type => linear_quad_type)
+                            elseif occursin(quadratic_hexes, current_element_type)
+                                linear_hex_type = "C3D8"
+                                new_line = replace(line,
+                                                   current_element_type => linear_hex_type)
+                            end
+                            println(outfile, new_line)
+                        end
+                    else # Element data in line
+                        # Check for linear elements - then we just copy the line
+                        if occursin(linear_quads, current_element_type) ||
+                           occursin(linear_hexes, current_element_type)
+                            println(outfile, line)
+                        else
+                            parts = split(line, ',')
+                            if occursin(quadratic_quads, current_element_type)
+                                # Print the first (element), second to fifth (vertices 1-4) indices to file.
+                                # For node order of quadratic (second-order) quads, see
+                                # http://130.149.89.49:2080/v2016/books/usb/default.htm?startat=pt06ch28s01ael02.html
```

Comment:
Is this the ABAQUS user guide?

Reply:
Yes, [user guide/documentation](http://130.149.89.49:2080/v2016/index.html). Unfortunately, I could not find the docs of more recent versions open accessible.